futures-core/src/task_impl/mod.rs

```diff
@@ -218,57 +217,16 @@ impl<T: ?Sized> Spawn<T> {
         where N: Clone + Into<NotifyHandle>,
               T: Stream,
     {
-        let mk = || notify.clone().into();
-        self.enter(BorrowedUnpark::new(&mk, id), |s| s.poll())
-    }
-
-    /// Invokes the underlying `start_send` method with this task in place.
-    ///
-    /// If the underlying operation returns `NotReady` then the `notify` value
-    /// passed in will receive a notification when the operation is ready to be
-    /// attempted again.
-    pub fn start_send_notify<N>(&mut self,
-                                value: T::SinkItem,
-                                notify: &N,
-                                id: usize)
-                               -> StartSend<T::SinkItem, T::SinkError>
-        where N: Clone + Into<NotifyHandle>,
-              T: Sink,
-    {
-        let mk = || notify.clone().into();
-        self.enter(BorrowedUnpark::new(&mk, id), |s| s.start_send(value))
+        self.poll_notify(notify, id, |s| s.poll())
     }
 
-    /// Invokes the underlying `poll_complete` method with this task in place.
-    ///
-    /// If the underlying operation returns `NotReady` then the `notify` value
-    /// passed in will receive a notification when the operation is ready to be
-    /// attempted again.
-    pub fn poll_flush_notify<N>(&mut self,
-                                notify: &N,
-                                id: usize)
-                                -> Poll<(), T::SinkError>
-        where N: Clone + Into<NotifyHandle>,
-              T: Sink,
-    {
-        let mk = || notify.clone().into();
-        self.enter(BorrowedUnpark::new(&mk, id), |s| s.poll_complete())
-    }
-
-    /// Invokes the underlying `close` method with this task in place.
-    ///
-    /// If the underlying operation returns `NotReady` then the `notify` value
-    /// passed in will receive a notification when the operation is ready to be
-    /// attempted again.
-    pub fn close_notify<N>(&mut self,
-                           notify: &N,
-                           id: usize)
-                           -> Poll<(), T::SinkError>
+    /// Invokes the function with the provided `notify` in the task context.
+    pub fn poll_notify<N, F, R>(&mut self, notify: &N, id: usize, f: F) -> R
```

Comment:
Just to confirm, was this intended to be `pub`? (not sure if it needs to be)

Reply:
Yeah-- I was trying to leave a way for people to `poll` a `Sink` now that this crate doesn't include the `Sink` trait.